crates/provider-keyvalue-redis/src/lib.rs

```diff
@@ -317,6 +434,61 @@ impl keyvalue::batch::Handler<Option<Context>> for KvRedisProvider {
     }
 }
 
+impl keyvalue::watcher::Handler<Option<Context>> for KvRedisProvider {
+    async fn on_set(
+        &self,
+        cx: Option<Context>,
```

Comment:
I think this logic should go under `receive_link_config` since that's where you'll want to setup your watches